fish-rust/src/util.rs

```diff
@@ -80,15 +81,19 @@ pub fn wcsfilecmp(a: wcharz_t, b: wcharz_t) -> i32 {
         acl = acl.to_uppercase().next().unwrap();
         bcl = bcl.to_uppercase().next().unwrap();
 
-        if acl < bcl {
-            retval = -1;
-            break;
-        } else if acl > bcl {
-            retval = 1;
-            break;
-        } else {
-            ai += 1;
-            bi += 1;
+        match acl.cmp(&bcl) {
+            Ordering::Less => {
+                retval = -1;
+                break;
+            }
+            Ordering::Equal => {
+                ai += 1;
+                bi += 1;
+            }
+            Ordering::Greater => {
```

Comment:
I discarded this variant because it seemed worse (more lines).
But I didn't realize it's a clippy lint, so this fix is fine.

We could already rewrite the entire function to return `Ordering`, only the ffi shim needs to return `c_int`.

Reply:
I think the proposed version is not really better than the old one and the old one matches the old code (w.r.t. order).
How about we disable this warning at crate-level?
We can re-enable it once we move to more idiomatic Rust.

Reply:
I spent a couple minutes rewriting it to use `Ordering` instead, let me know what you think. I've added it as a separate commit to make it easier to review or revert.